posthog/models/historical_version.py

```diff
@@ -0,0 +1,120 @@
+import copy
+from datetime import datetime
+from typing import Dict
+
+from django.db import models
+from django.utils import timezone
+
+
+def as_deletion_state(metadata: Dict) -> Dict:
+    """
+    Deletion state is being saved without access to a DRF serializer
+    So replace datetime field with a string or it can't be serialized to JSON
+    """
```

Comment:
If this is purely about JSON encoding, a more lightweight approach (avoiding `deepcopy`) would be using `json.dumps`'s `default` kwarg. See how `posthog_app_context` is encoded in a way that supports UUID values – the same could be used for `datetime`:
```Python
json.dumps(posthog_app_context, default=json_uuid_convert)
```

Reply:
Which led me to discover that the django JSONField can take a json encoder 💯